machine/v1beta1/types_powervsprovider.go

```diff
@@ -0,0 +1,156 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PowerVSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for a PowerVS virtual machine. It is used by the PowerVS machine actuator to create a single Machine.
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PowerVSMachineProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// UserDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance.
+	// +optional
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// CredentialsSecret is a reference to the secret with IBM Cloud credentials.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// Region indicates the Power VS ServiceInstance region.
+	// +optional
+	Region string `json:"region,omitempty"`
+
+	// Zone indicates the Power VS ServiceInstance zone.
+	// +optional
+	Zone string `json:"zone,omitempty"`
+
+	// ServiceInstance is the reference to the Power VS ServiceInstance on which the machine instance will be created.
+	// +kubebuilder:validation:=Required
+	ServiceInstance PowerVSResource `json:"serviceInstance"`
+
+	// Image is the OS image to use to create the instance.
+	// +kubebuilder:validation:=Required
+	Image PowerVSResource `json:"image"`
+
+	// Network is the reference to the Network to use for this instance.
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// KeyPairName is the name of the KeyPair to use for SSH
+	// +kubebuilder:validation:=Required
+	KeyPairName string `json:"keyPairName"`
+
+	// SysType is the System type used to host the instance.
+	// Defaults to s922.
+	// +optional
+	SysType string `json:"sysType,omitempty"`
+
+	// ProcessorType is the VM instance processor type.
+	// It must be set to one of the following values: dedicated, shared or capped.
+	// Defaults to shared.
+	// +kubebuilder:validation:Enum:=dedicated;shared;capped
+	// +optional
+	ProcType string `json:"procType,omitempty"`
+
+	// Processors is the number of virtual processors in a virtual machine.
+	// Defaults to 0.5.
+	// +optional
+	Processors string `json:"processors,omitempty"`
+
+	// Memory is the size of a virtual machine's memory, in GB.
+	// Defaults to 32 GB.
+	// +optional
+	Memory int32 `json:"memory,omitempty"`
```

Comment:
Typically we use resource.Quantity for something like this and allow the user to choose the units. In the operator itself you can validate that it meets the requirements. 

In this case though, it may not make sense if the only value is a `GiB` (we are talking GiB and not GB right?), in which case perhaps you want to include the unit in the field name?

Reply:
yes, this is in `GiB`, you mean something like

```suggestion
	// Memory is the size of a virtual machine's memory, in GB.
	// Defaults to 32 GB.
	// +optional
	MemoryGiB int32 `json:"memoryGiB,omitempty"`
```

Reply:
Yeah I think that makes it clearer that this field does not need to have a string unit attached